### DISCUSSION TITLE
I can't get rid of the dark diamonds

### DISCUSSION THREAD

**Discussion by Kelevra142** [JanDeDobbeleer/oh-my-posh#5789] [Troubleshoot] [answered]
I tried to make those dark diamonds transparent for a few hours without success. A google search brought up a thread from a few years ago that stated that this was bugged on Windows when using a transparent background.

I am on Linux using KDE Plasma 6 with a transparent background, is there any way around this or is this also bugged?
![image](https://github.com/user-attachments/assets/714a9ea7-2b51-4d1a-9fe8-489680538a4f)

This is my config:

{
  "$schema": "https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json",
  "blocks": [
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "background": "#f4492c",
          "foreground": "#575656",
          "leading_diamond": "\ue0b2",
          "style": "diamond",
          "template": "\uf489 {{ .Name }}",
          "trailing_diamond": "<transparent,#f4492c>\ue0b2</>",
          "type": "shell"
        },
        {
          "background": "#8c01f3",
          "foreground": "#FEF5ED",
          "leading_diamond": "\ue0b2",
          "style": "diamond",
          "template": "RAM: {{ round .PhysicalPercentUsed .Precision }} \uf295 | {{ (div ((sub .PhysicalTotalMemory .PhysicalAvailableMemory)|float64) 1073741824.0) }}/{{ (div .PhysicalTotalMemory 1073741824.0) }}GB \uefc5 ",
          "trailing_diamond": "<transparent,#8c01f3>\ue0b2</>",
          "type": "sysinfo"
        },
        {
          "background": "#f4d017",
          "foreground": "#575656",
          "leading_diamond": "\ue0b2",
          "properties": {
            "style": "roundrock",
            "threshold": 0
          },
          "style": "diamond",
          "template": " {{ .FormattedMs }} ",
          "trailing_diamond": "\ue0b0",
          "type": "executiontime"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "right",
      "segments": [
        {
          "background": "#17D7A0",
          "foreground": "#011627",
          "leading_diamond": "\ue0b2",
          "properties": {
            "branch_icon": "\ue725 ",
            "fetch_stash_count": true,
            "fetch_status": true,
            "fetch_upstream_icon": true,
            "fetch_worktree_count": true
          },
          "style": "diamond",
          "template": " {{ .UpstreamIcon }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} \uf044 {{ .Working.String }}{{ end }}{{ if and (.Working.Changed) (.Staging.Changed) }} |{{ end }}{{ if .Staging.Changed }} \uf046 {{ .Staging.String }}{{ end }}{{ if gt .StashCount 0 }} \ueb4b {{ .StashCount }}{{ end }} ",
          "trailing_diamond": "\ue0b0",
          "type": "git"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "style": "plain",
          "template": "\u256d\u2500",
          "type": "text"
        },
        {
          "properties": {
            "time_format": "15:04:05"
          },
          "style": "plain",
          "template": " \uf017 {{ .CurrentDate | date .Format }} |",
          "type": "time"
        },
        {
          "style": "plain",
          "template": " \uf292 ",
          "type": "root"
        },
        {
          "properties": {
            "folder_icon": "\uf07b ",
            "folder_separator_icon": " \uf061 ",
            "home_icon": "\uf015 "
          },
          "style": "plain",
          "template": " {{ .Path }} ",
          "type": "path"
        }
      ],
      "type": "prompt"
    },
    {
      "alignment": "left",
      "newline": true,
      "segments": [
        {
          "properties": {
            "always_enabled": true
          },
          "style": "plain",
          "template": "\u2570\u2500 ",
          "type": "status"
        }
      ],
      "type": "prompt"
    }
  ],
  "console_title_template": "{{ .Folder }}",
  "transient_prompt": {
    "background": "transparent",
    "foreground": "#FEF5ED",
    "template": "\ue285 "
  },
  "version": 2
}

**Comment by Kelevra142** [comment, marked as answer]
Alright, after a full nights sleep I got back to it and tried a few more times.
The solution is quite easy:

    "trailing_diamond": "<transparent,#f4492c>\ue0b2</>",

Remove the Attributes that alter the symbols design and use the correct symbol so it looks like this:

    "trailing_diamond": "<\ue0d6</>",

Which returns this result:

![image](https://github.com/user-attachments/assets/864aeb6d-4cf0-447c-8a63-b782a43ccb13)

Right side shows the previous outcome, left side the desired result.

**Reply by JanDeDobbeleer** [reply]
@Kelevra142 the issue is that the terminal doesn't support using inverted ANSI. So it prints black rather than transparent. You can resort to using the [inverted powerline glyphs](https://github.com/ryanoasis/nerd-fonts/pull/1490) as well. That's available in the latest NF version(s). You can make use of `leading_powerline_symbol` here to get it absolutely right.

**Reply by Kelevra142** [reply]
Thank you for your reply!
I got it fixed and then changed the design, but I will continue to experiment with powerline.
